### DISCUSSION TITLE
Lambda/serverless functions support

### DISCUSSION THREAD

**Discussion by ChristopherBiscardi** [planetscale/discussion#32] [Feature Requests] [answered]
It seems that the only way to connect to a pscale database is to use the CLI (`pscale shell` or `pscale connection`). 

Is it a goal to support lambda and lambda-like environments? (netlify/etc functions, cloudflare workers, etc)

**Comment by samlambert** [comment]
Hey Chris!

Thank you for reaching out. We are working on expanding our client support which will enable you to use functions.

Thanks,
Sam

**Comment by adrians5j** [comment]
Another question I have regarding the Lambda/serverless functions support.

It seems we need to establish a connection first, in order to interact with the database. Based on past experiences, this model does not work great with serverless, more specifically, AWS Lambda, simply because multiple Lambda function instances create multiple database connections. Ultimately, we reach the total number of connections, and we end up with function instances not being able to connect.

In the future, will there be a way to not-connect, and simply execute database operations via HTTP requests (similar to DynamoDB maybe)? 

I'm sorry if there's already a way to do this. :)

Thanks in advance :)

**Reply by mscoutermarsh** [reply]
Hi @doitadrian,

PlanetScale can handle 10s of thousands of connections per database branch. So that you don't have to worry about the typical connection limits that you've had when using MySQL/Postgres etc.

**Reply by adrians5j** [reply]
Thanks @mscoutermarsh, that's very cool 🙂

**Reply by yeahse** [reply]
@mscoutermarsh Hi, Are there still only approaches via EC2 or Lambda?

**Reply by mscoutermarsh** [reply]
What platform are you trying to use?

**Reply by yeahse** [reply]
> What platform are you trying to use?

I use Flutter

**Reply by mscoutermarsh** [reply]
@yeahse I haven't personally used it. But from what I can see, Flutter does work with MySQL. Since we're MySQL compatible, it should work with PlanetScale.

**Comment by mscoutermarsh** [comment, marked as answer]
Update: we've now made serverless support simple. You can [generate a connection string](https://docs.planetscale.com/concepts/connection-strings) and then use it with any MySQL client. No need to worry about connection limits, PlanetScale can handle tens of thousands of simultaneous database connections.

This works for any serverless platform that supports tcp connections (vercel, lambda, azure, google etc). Cloudflare workers are currently not supported as they only support HTTP connections. We're looking into how we can support the in the future.

**Reply by matteolobello** [reply]
Any updates on Cloudflare Workers support?

**Reply by mscoutermarsh** [reply]
@matteolobello Nothing yet, we haven't forgotten about it though. 😄

**Reply by gregmsanderson** [reply]
Cloudflare Workers support would be great. Seems like to make that work you would need to support being able to pass a query in a HTTP request (REST API style). Like CosmosDB, DynamoDB, and friends. Workers are _planning_ TCP connections but currently that requires a Cloudflare Tunnel on the database-end. They have some blog posts on that in the past couple of days. But it _seems_ the Tunnel needs to be linked to an account (like if you have your own database server). So not sure that approach works with a shared/cloud hosted service.

**Reply by ydennisy** [reply]
@mscoutermarsh I am having issues with connection using the connection string.

I have a TS app using the Prisma client and deployed to Google Cloud Run.

I have used the connection string as provided for Prisma from the UI, and I am receiving the following error:
```
Error opening a TLS connection: error:1416F086:SSL
routines:tls_process_server_certificate:certificate verify
failed:../deps/openssl/openssl/ssl/statem/statem_clnt.c:1916: (unable to get
local issuer certificate)
```

Do you know if anything else needs to be installed on the machine or configured in the connection string to get this working?

It seems that Prisma cannot find the local certs?

Setting `sslaccept=accept_invalid_certs` works - but seems not ideal!

**Reply by jgreet** [reply]
@ydennisy perhaps the `ca-certificates` package isn't installed in your image? If that isn't it, could you share the Docker image you're starting from and your Node and Prisma versions? Feel free to send those details to support@planetscale.com if you prefer.
